src/dune_engine/action.ml

```diff
@@ -227,6 +229,16 @@ let link_function ~(mode : Sandbox_mode.some) : path -> target -> t =
       fun a b ->
     Symlink (a, b)
   | Copy -> fun a b -> Copy (a, b)
+  | Hardlink ->
+    if Sys.win32 then
```

Comment:
At first sight, it seems that hard links *are* supported under Windows; is there a specific aspect that makes them not suitable?

Reply:
The first implementation takes a conservative approach. Hard links may indeed be supported on Windows but I remember hearing that using them requires non-standard user permissions. If it turns out that we can enable hard links on Windows that would be great, and easy to do.

Reply:
Got it, thanks!

Reply:
FTR, in the shared cache we detect whether hardlinks are supported by trying to do a hardlink: https://github.com/ocaml/dune/blob/0693738306e5cbcbc4d64ea68ead9fbe650d32ea/src/cache/local.ml#L345-L358

We could centralise this test.

Reply:
Agreed, that seems useful.

Reply:
It seems best to keep this PR simple, so I added a comment about implementing this functionality in future, when we have a bit more time to think it through.

Reply:
I created an issue to track this: https://github.com/ocaml/dune/issues/4362